chia/full_node/weight_proof_v2.py

```diff
@@ -0,0 +1,1758 @@
+from chia.consensus.vdf_info_computation import get_signage_point_vdf_info
+from chia.types.blockchain_format.classgroup import ClassgroupElement, B
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.types.blockchain_format.slots import ChallengeChainSubSlot, RewardChainSubSlot, ChallengeBlockInfo
+from chia.types.blockchain_format.sub_epoch_summary import SubEpochSummary
+from chia.types.blockchain_format.vdf import (
+    VDFInfo,
+    compress_output,
+    verify_compressed_vdf,
+)
+from chia.types.end_of_slot_bundle import EndOfSubSlotBundle
+from chia.types.header_block import HeaderBlock
+from chia.types.weight_proof import (
+    RecentChainData,
+    SubEpochChallengeSegmentV2,
+    SubEpochData,
+    SubEpochSegmentsV2,
+    SubSlotDataV2,
+    WeightProofV2,
+)
+from chia.util.block_cache import BlockCache
+from chia.util.hash import std_hash
+from chia.util.ints import uint8, uint32, uint64, uint128
+from chia.util.streamable import recurse_jsonify, dataclass_from_dict
+
+log = logging.getLogger(__name__)
+
+
+class WeightProofHandlerV2:
+    LAMBDA_L = 100
+    C = 0.5
+    MAX_SAMPLES = 140
+
+    def __init__(
+        self,
+        constants: ConsensusConstants,
+        blockchain: BlockchainInterface,
+    ):
+        self.tip: Optional[bytes32] = None
+        self.proof: Optional[WeightProofV2] = None
+        self.constants = constants
+        self.blockchain = blockchain
+        self.lock = asyncio.Lock()
+
+    async def get_proof_of_weight(self, tip: bytes32, seed: bytes32) -> Optional[WeightProofV2]:
+
+        tip_rec = self.blockchain.try_block_record(tip)
+        if tip_rec is None:
+            log.error("unknown tip")
+            return None
+
+        if tip_rec.height < self.constants.WEIGHT_PROOF_RECENT_BLOCKS:
+            log.debug("need at least 3 sub epochs for weight proof")
+            return None
+
+        async with self.lock:
+            wp = await self._create_proof_of_weight(tip, seed)
+            if wp is None:
+                return None
+            self.proof = wp
+            self.tip = tip
+            return wp
+
+    def get_fork_point_no_validations(self, weight_proof: WeightProofV2) -> Tuple[bool, uint32]:
+        log.debug("get fork point skip validations")
+        if len(weight_proof.sub_epochs) == 0:
+            return False, uint32(0)
+        summaries, sub_epoch_weight_list = _validate_sub_epoch_summaries(self.constants, weight_proof)
+        if summaries is None:
+            log.warning("weight proof failed to validate sub epoch summaries")
+            return False, uint32(0)
+        return True, self.get_fork_point(summaries)
+
+    async def validate_weight_proof(
+        self, weight_proof: WeightProofV2, seed: bytes32, skip_segments: bool = False
+    ) -> Tuple[bool, uint32, List[SubEpochSummary], List[BlockRecord]]:
+        valid, summaries, records = await validate_weight_proof_no_fork_point(
```

Comment:
Why do we have this method of validation without fork point? Is getting the fork point slow? If not, I think it's better that have less methods

Reply:
i did this to be compatible to the wallet 
https://github.com/Chia-Network/chia-blockchain/blob/9f48b5ac6c4899179498f076908739bcf6596078/chia/wallet/wallet_node.py#L1147

ill switch to the  normal one and just not take the fork_point